Jenkinsfile

```diff
@@ -87,13 +87,9 @@ def getBuildContext() {
   def profile = "${params.PROFILE}"
   def EVENT_LOOP = "${params.EVENT_LOOP.toLowerCase()}"
   matrixType = "SMOKE"
-  developBranchPattern = ~"((dev|long)-)?python-.*"
 
-  if (developBranchPattern.matcher(env.BRANCH_NAME).matches()) {
-    matrixType = "DEVELOP"
-    if (env.BRANCH_NAME.contains("long")) {
-      profile = "FULL"
-    }
+  if (env.BRANCH_NAME.contains("long")) {
```

Comment:
Should we remove this `if` as well?

Reply:
That's an interesting question.

There's going to be some changes to all these build matrices to support our new version support policy for the Python driver; from the new release on a given driver release will only support Python versions that aren't out-of-support at the time of release.  So 2.7, 3.5 and 3.6 will be removed from all five build matrices, for instance.  There's also a legit question for me about whether we really need five distinct build matrices... that seems like a lot.

But as I thought about it you're actually asking a different question; if we're already moving away from branch name correlation to a build matrix why not sever that tie completely?  I think this is prolly right; I don't really like the idea that a branch name magically correlates to a specific build matrix.  Seems better to maybe run other matrices on regular schedules and support ad-hoc runs if somebody really wants to run something now.

Whaddya think?

Reply:
> if we're already moving away from branch name correlation to a build matrix why not sever that tie completely

Exactly, if a branch gets named "something-long" and the build starts failing we may take a lot of time to realize it's using a completely different build because of this `if`, I'd just remove it.

Reply:
Sold.  I was originally avoiding making this PR too big under the assumption that the overall build matrix would have to be revisited once we change supported versions of the Python runtime but as discussed above that concern isn't explicitly connected to what you're advocating here @joao-r-reis .

I'll make the change.